lib/transmuxer/mpeg_ts_transmuxer.js

```diff
@@ -0,0 +1,173 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.transmuxer.MpegTsTransmuxer');
+
+goog.require('shaka.media.Capabilities');
+goog.require('shaka.transmuxer.MpegAudio');
+goog.require('shaka.transmuxer.TransmuxerEngine');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.Error');
+goog.require('shaka.util.ManifestParserUtils');
+goog.require('shaka.util.MimeUtils');
+goog.require('shaka.util.TsParser');
+goog.require('shaka.util.Uint8ArrayUtils');
+
+
+/**
+ * @implements {shaka.extern.Transmuxer}
+ * @export
+ */
+shaka.transmuxer.MpegTsTransmuxer = class {
+  /**
+   * @param {string} mimeType
+   */
+  constructor(mimeType) {
+    /** @private {string} */
+    this.originalMimeType_ = mimeType;
+
+    /** @private {?shaka.util.TsParser} */
+    this.tsParser_ = null;
+  }
+
+
+  /**
+   * @override
+   * @export
+   */
+  destroy() {
+    // Nothing
+  }
+
+
+  /**
+   * Check if the mime type and the content type is supported.
+   * @param {string} mimeType
+   * @param {string=} contentType
+   * @return {boolean}
+   * @override
+   * @export
+   */
+  isSupported(mimeType, contentType) {
+    const Capabilities = shaka.media.Capabilities;
+
+    if (!this.isTsContainer_(mimeType)) {
+      return false;
+    }
+
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+    const MimeUtils = shaka.util.MimeUtils;
+
+    const codecs = MimeUtils.getCodecs(mimeType);
+    const allCodecs = MimeUtils.splitCodecs(codecs);
+
+    const audioCodec = shaka.util.ManifestParserUtils.guessCodecsSafe(
+        ContentType.AUDIO, allCodecs);
+    const videoCodec = shaka.util.ManifestParserUtils.guessCodecsSafe(
+        ContentType.VIDEO, allCodecs);
+
+    if (!audioCodec || videoCodec) {
+      return false;
+    }
+    const normalizedCodec = MimeUtils.getNormalizedCodec(audioCodec);
+
+    if (normalizedCodec != 'mp3') {
+      return false;
+    }
+
+    return Capabilities.isTypeSupported(
+        this.convertCodecs(ContentType.AUDIO, mimeType));
+  }
+
+
+  /**
+   * Check if the mimetype is 'video/mp2t'.
+   * @param {string} mimeType
+   * @return {boolean}
+   * @private
+   */
+  isTsContainer_(mimeType) {
+    return mimeType.toLowerCase().split(';')[0] == 'video/mp2t';
+  }
+
+
+  /**
+   * @override
+   * @export
+   */
+  convertCodecs(contentType, mimeType) {
```

Comment:
This transmuxer is very specialized yes, I admit it. This transmuxer takes an audio-only TS with MP3, and converts it to raw MP3(`audio/mpeg`). We don't do it in ts_transmuxer.js because the output of it is always MP4.
Why do I do this so specialized? Because the only browser that supports MP3 in MP4 is Firefox(`audio/mp4; codecs="mp4a.40.34"`), other browsers don't support it.
But is having this type of stream common? In Europe, many radio stations that broadcast on DVB-T use MP3 within TS, and having this converter saves having to convert codecs within the headend.

Reply:
Okay, sounds good.  Could you add comments to explain that?  Either under `@fileoverview` at the top of the file, or on the class definition itself.  Thanks!

Reply:
Added!